Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp

```diff
@@ -436,60 +415,62 @@ namespace AzToolsFramework::Prefab
         }
     }
 
-    void PrefabFocusHandler::OpenInstanceContainers(const AZStd::vector<AZ::EntityId>& instances) const
+    void PrefabFocusHandler::SetInstanceContainersOpenState(const AZStd::vector<RootAliasPath>& instances, bool openState) const
     {
         // If this is called outside the Editor, this interface won't be initialized.
         if (!m_containerEntityInterface)
         {
             return;
         }
-        
-        for (const AZ::EntityId& containerEntityId : instances)
+
+        for (const RootAliasPath& rootAliasPath : instances)
         {
-            InstanceOptionalReference instance = GetReferenceFromContainerEntityId(containerEntityId);
+            InstanceOptionalReference instance = GetInstanceReferenceFromRootAliasPath(rootAliasPath);
 
             if (instance.has_value())
             {
-                m_containerEntityInterface->SetContainerOpen(instance->get().GetContainerEntityId(), true);
+                m_containerEntityInterface->SetContainerOpen(instance->get().GetContainerEntityId(), openState);
             }
         }
     }
 
-    void PrefabFocusHandler::CloseInstanceContainers(const AZStd::vector<AZ::EntityId>& instances) const
+    InstanceOptionalReference PrefabFocusHandler::GetInstanceReferenceFromRootAliasPath(RootAliasPath rootAliasPath) const
     {
-        // If this is called outside the Editor, this interface won't be initialized.
-        if (!m_containerEntityInterface)
-        {
-            return;
-        }
+        PrefabEditorEntityOwnershipInterface* prefabEditorEntityOwnershipInterface =
+            AZ::Interface<PrefabEditorEntityOwnershipInterface>::Get();
 
-        for (const AZ::EntityId& containerEntityId : instances)
+        if (prefabEditorEntityOwnershipInterface)
         {
-            InstanceOptionalReference instance = GetReferenceFromContainerEntityId(containerEntityId);
+            InstanceOptionalReference instance = prefabEditorEntityOwnershipInterface->GetRootPrefabInstance();
 
-            if (instance.has_value())
+            int i = 0;
```

Comment:
Suggestion: My understanding is that `i` is being used to determine whether or not `pathElement` is the first (root) element of `rootAliasPath`. Is there not a way we can do this in an easier to read manner? I.e. compare `pathElement` with the beginning of the `rootAliasPath` (i.e. `rootAliasPath.begin()`, I don't actually know what type of container this is, but something equivalent)?